packages/client/src/components/atoms/AddToCalendar/AddToCalendar.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { useTranslation, ActionButton } from "@eisbuk/translations";
+import { useSelector } from "react-redux";
+import { DateTime } from "luxon";
+import { ICalendar } from "datebook";
+
+import makeStyles from "@mui/styles/makeStyles";
+
+import Button from "@mui/material/Button";
+
+import { LocalStore } from "@/types/store";
+import { __addToCalendarButtonId__ } from "@/__testData__/testIds";
+import { getAboutOrganization } from "@/store/selectors/app";
+
+interface Props {
+  /**
+   * Record of subscribed slots with subscribed slotIds as keys and subscribed duration as value.
+   * Doesn't need to be organized as we're checking for each value by key (no need for ordering and grouping).
+   */
+  bookedSlots?: LocalStore["firestore"]["data"]["bookedSlots"];
+}
+const AddToCalendar: React.FC<Props> = ({ bookedSlots = {} }) => {
+  const classes = useStyles();
+
+  const { displayName = "", location = "" } =
+    useSelector(getAboutOrganization) || {};
+  const { t } = useTranslation();
+
+  // is displayed when bookings are finalized aka deadline is null
+  const handleClick = () => {
+    let icalendar: ICalendar;
+
+    Object.values(bookedSlots).forEach((bookedSlot, i, originalArr) => {
+      const startDate = getStartDate(bookedSlot.date, bookedSlot.interval);
+      const endDate = getEndDate(bookedSlot.date, bookedSlot.interval);
+
+      const bookedSlotEvent = {
+        title: `Booked Slot at ${displayName}`,
+        location: location,
+        start: startDate,
+        end: endDate,
+      };
+
+      /**
+       * ICalendar needs to be instantiated with an event
+       * which can't be an empty object so on the first iteration
+       * it's instantiated with the first bookedSlotEvent
+       */
+      if (i === 0) {
+        icalendar = new ICalendar(bookedSlotEvent).addProperty(
+          "UID",
+          `${bookedSlot.date}${bookedSlot.interval}`
+        );
+      } else {
+        /**
+         * To add another event to the calendar it must be of type ICalendar
+         */
+        icalendar
+          .addEvent(new ICalendar(bookedSlotEvent))
+          .addProperty("UID", `${bookedSlot.date}${bookedSlot.interval}`);
+      }
+      /**
+       * Download calendar on last iteration
+       */
+      if (i === originalArr.length - 1) {
+        icalendar.download("Booked_Slots.ics");
+      }
```

Comment:
Can we just download the calendar after the loop is finished (that way there's no need to comment/explain anything there)

Reply:
it's getting instantiated inside the loop though so it couldn't be used outside @ikusteu

Reply:
It’s instantiated in the loop, but is declared outside the loop, therefore, it’s accessible from outer scope of the loop. I don’t know why TypeScript complains about it…but you can set it to an empty object when you declare it and cast the type, like so: ‘let calendar = {} as ICalendar’